cfgmgr/buffermgrdyn.cpp

```diff
@@ -0,0 +1,1674 @@
+                }
+
+                SWSS_LOG_NOTICE("Buffer pool %s had been updated with new size [%s]", poolName.c_str(), pool.total_size.c_str());
+            }
+            else
+            {
+                SWSS_LOG_INFO("Buffer pool debug info %s", i.c_str());
+            }
+        }
+    }
+    catch (...)
+    {
+        SWSS_LOG_WARN("Lua scripts for buffer calculation were not executed successfully");
+    }
+}
+
+void BufferMgrDynamic::checkSharedBufferPoolSize()
+{
+    // PortInitDone indicates all steps of port initialization has been done
+    // Only after that does the buffer pool size update starts
+    if (!m_portInitDone)
+    {
+        vector<FieldValueTuple> values;
+        if (m_applPortTable.get("PortInitDone", values))
+        {
+            SWSS_LOG_NOTICE("Buffer pools start to be updated");
+            m_portInitDone = true;
+        }
+        else
+        {
+            if (m_firstTimeCalculateBufferPool)
+            {
+                // It's something like a placeholder especially for warm reboot flow
+                // without all buffer pools created, buffer profiles are unable to be cureated,
+                // which in turn causes buffer pgs and buffer queues unable to be created,
+                // which prevents the port from being ready and eventually fails the warm reboot
+                // After the buffer pools are created for the first time, we won't touch it
+                // until portInitDone
+                // Eventually, the correct values will pushed to APPL_DB and then ASIC_DB
+                recalculateSharedBufferPool();
+                m_firstTimeCalculateBufferPool = false;
+                SWSS_LOG_NOTICE("Buffer pool update defered because port is still under initialization, start polling timer");
+            }
+
+            return;
+        }
+    }
+
+    if (!m_mmuSize.empty())
+        recalculateSharedBufferPool();
+}
+
+// For buffer pool, only size can be updated on-the-fly
+void BufferMgrDynamic::updateBufferPoolToDb(const string &name, const buffer_pool_t &pool)
+{
+    vector<FieldValueTuple> fvVector;
+
+    if (pool.ingress)
+        fvVector.emplace_back(make_pair("type", "ingress"));
+    else
+        fvVector.emplace_back(make_pair("type", "egress"));
+
+    fvVector.emplace_back(make_pair("mode", pool.mode));
+
+    SWSS_LOG_INFO("Buffer pool %s is initialized", name.c_str());
+
+    fvVector.emplace_back(make_pair("size", pool.total_size));
+
+    m_applBufferPoolTable.set(name, fvVector);
+
+    m_stateBufferPoolTable.set(name, fvVector);
+}
+
+void BufferMgrDynamic::updateBufferProfileToDb(const string &name, const buffer_profile_t &profile)
+{
+    vector<FieldValueTuple> fvVector;
+    string mode = getPgPoolMode();
+
+    if (mode.empty())
+    {
+        // this should never happen if switch initialized properly
+        SWSS_LOG_ERROR("PG lossless pool is not yet created, creating profile %s failed", name.c_str());
+        return;
```

Comment:
Actually, this check can be removed because it is called from two functions:
- `allocateProfile`, in which there is already a check ahead of calling
- `doUpdateStaticProfileTask`, which is triggered by CLI (or other management plane interfaces) and the ingress_lossless_pool has been created before the CLI ready.